app/services/clients/connector.client.js

```diff
@@ -167,6 +173,34 @@ ConnectorClient.prototype = {
         .on('error', callbackToPromiseConverter)
     })
   },
+  /**
+   * Retrieves gateway account by external ID
+   * @param params
+   *          An object with the following elements;
+   *            gatewayAccountExternalId (required)
+   *            correlationId (optional)
+   *@return {Promise}
+   */
+  getAccountByExternalId: function (params) {
+    return new Promise((resolve, reject) => {
+      let url = _accountByExternalIdUrlFor(params.gatewayAccountExternalId, this.connectorUrl)
+      let startTime = new Date()
+      let context = {
+        url: url,
+        defer: { resolve: resolve, reject: reject },
+        startTime: startTime,
+        correlationId: params.correlationId,
+        method: 'GET',
+        description: 'get an account by external id',
+        service: SERVICE_NAME
+      }
+
+      let callbackToPromiseConverter = createCallbackToPromiseConverter(context)
+
+      oldBaseClient.get(url, params, callbackToPromiseConverter, null)
```

Comment:
If possible it would be ideal if we didn't add any new instances of using the old base client, I think Connector already has examples of methods that use the new base client? 

This would
* avoid adding new promise construction code
* allow upstream code to use async/ await patterns with no additional effort
* not require a connector client instance per request (for the pub/ sub error handling)

Reply:
There are few methods in connector client using new base client. Updated!